modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -103,6 +135,17 @@ private boolean doBinaryComparisonCoercion(SqlCallBinding binding) {
         RelDataType leftType = binding.getOperandType(0);
         RelDataType rightType = binding.getOperandType(1);
 
+        if (SqlTypeUtil.isIntType(leftType) && SqlTypeUtil.isCharacter(rightType)
+                || SqlTypeUtil.isIntType(rightType) && SqlTypeUtil.isCharacter(leftType)) {
+
+            SqlLiteral lit = extractLiteral(call, validator);
+
+            if (lit != null) {
+                String strVal = lit.toValue();
+                checkStringContainDigitsOnly(strVal, binding.getCall().getParserPosition());
+            }
+        }
+
         //
```

Comment:
This PR updates `binaryComparisonCoercion` but `binaryArithmeticCoercion` remains unchanged why is that?